src/freenet/clients/http/UserAlertsToadlet.java

```diff
@@ -47,7 +50,16 @@ public void handleMethodPOST(URI uri, HTTPRequest request, ToadletContext ctx) t
 			int userAlertHashCode = request.getIntPart("disable", -1);
 			ctx.getAlertManager().dismissAlert(userAlertHashCode);
 		}
-		headers.put("Location", ".");
+		
+		
+		String redirect;
+		try {
+			redirect = request.getPartAsStringThrowing("redirectToAfterDisable", 1024);
+		} catch (SizeLimitExceededException | NoSuchElementException e) {
+			redirect = ".";
+		}
+		headers.put("Location", redirect);
```

Comment:
@xor-freenet could you have a stab at fixing this? It is from https://github.com/freenet/fred/pull/607/commits/e2a63110e403f71a7596b049495eb27e72a60f82